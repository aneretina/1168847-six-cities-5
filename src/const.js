export const ZOOM = 12;

export const TO_PERCENT = 20;

export const CITIES = [
  `Amsterdam`,
  `Paris`,
  `Cologne`,
  `Hamburg`,
  `Dusseldorf`,
  `Brussels`
];


export const Icon = {
  URL: `/img/pin.svg`,
  ACTIVE_URL: `img/pin-active.svg`,
  SIZE: [30, 30],
};

export const ID_MAP_CONTAINER = `map`;

export const SortOptions = {
  POPULAR: `Popular`,
  PRICE_LOW_TO_HIGH: `Price: low to high`,
  PRICE_HIGH_TO_LOW: `Price: high to low`,
  TOP_RATED_FIRST: `Top rated first`
};

export const AuthorizationStatus = {
  NOT_AUTHORIZED: `NOT_AUTHORIZED`,
  AUTHORIZED: `AUTHORIZED`,
};

export const FavoriteBtnType = {
  PROPERTY: `property`,
  CARD: `place-card`
};

export const FavoriteBtnSize = {
  [FavoriteBtnType.PROPERTY]: {width: 31, height: 33},
  [FavoriteBtnType.CARD]: {width: 18, height: 19},
};

