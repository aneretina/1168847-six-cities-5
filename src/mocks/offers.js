export const offers = [
  {
    id: 15,
    city: {
      name: `Amsterdam`,
      location: {
        latitude: 52.38333,
        longitude: 4.9,
      }
    },
    location: {
      latitude: 52.3909553943508,
      longitude: 4.85309666406198,
    },
    pictures: [`img/apartment-01.jpg`, `img/apartment-01.jpg`, `img/apartment-01.jpg`, `img/apartment-01.jpg`, `img/apartment-01.jpg`, `img/apartment-01.jpg`],
    previewImage: `img/apartment-01.jpg`,
    isPremium: true,
    isFavorite: true,
    price: 180,
    title: `Wood and stone place`,
    type: `room`,
    rating: 3,
    description: `Discover daily local life in city center, friendly neighborhood, clandestine casino, karaoke, old-style artisans, art gallery and artist studio downstairs.`,
    bedroomsCount: 2,
    guestsLimit: 2,
    goods: [`Wi-Fi`, `Heating`, `Kitchen`, `Parking`, `Flowers`, `Dishwasher`, `Towels`, `TV`],
    host: {
      avatar: `img/avatar-max.jpg`,
      name: `Max`,
      isPro: false,
    }
  },

  {
    id: 2,
    city: {
      name: `Paris`,
      location: {
        latitude: 48.864716,
        longitude: 2.349014,
      }
    },
    location: {
      latitude: 48.87456,
      longitude: 2.413,
    },
    pictures: [`img/apartment-02.jpg`, `img/apartment-02.jpg`, `img/apartment-02.jpg`, `img/apartment-02.jpg`, `img/apartment-02.jpg`, `img/apartment-02.jpg`],
    previewImage: `img/apartment-02.jpg`,
    isPremium: true,
    isFavorite: true,
    price: 80,
    title: `Beautiful & luxurious apartment in Paris`,
    type: `Apartment`,
    rating: 4,
    description: `Discover daily local life in city center, friendly neighborhood, clandestine casino, karaoke, old-style artisans, art gallery and artist studio downstairs.`,
    bedroomsCount: 1,
    guestsLimit: 2,
    goods: [`Wi-Fi`, `Heating`, `Kitchen`, `Parking`, `Flowers`, `Dishwasher`, `Towels`, `TV`],
    host: {
      avatar: `img/avatar-maggy.jpg`,
      name: `Maggy`,
      isPro: true,
    }
  },

  {
    id: 9,
    city: {
      name: `Cologne`,
      location: {
        latitude: 50.935173,
        longitude: 6.953101,
      }
    },
    location: {
      latitude: 50.9682,
      longitude: 6.874,
    },
    pictures: [`img/apartment-03.jpg`, `img/apartment-03.jpg`, `img/apartment-03.jpg`, `img/apartment-03.jpg`, `img/apartment-03.jpg`, `img/apartment-03.jpg`],
    previewImage: `img/apartment-03.jpg`,
    isPremium: false,
    isFavorite: false,
    price: 220,
    title: `Beautiful & luxurious apartment in Cologne`,
    type: `Apartment`,
    rating: 1,
    description: `Discover daily local life in city center, friendly neighborhood, clandestine casino, karaoke, old-style artisans, art gallery and artist studio downstairs.`,
    bedroomsCount: 4,
    guestsLimit: 5,
    goods: [`Wi-Fi`, `Heating`, `Kitchen`, `Parking`, `Flowers`, `Dishwasher`, `Towels`, `TV`],
    host: {
      avatar: `img/avatar-bill.jpg`,
      name: `Bill`,
      isPro: true,
    }
  },
];
