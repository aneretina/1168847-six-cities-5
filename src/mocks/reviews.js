export default [
  {
    id: 1,
    name: `Jake`,
    avatar: [`https://placeimg.com/640/480/people/101`,
      `https://placeimg.com/640/480/people/102`,
      `https://placeimg.com/640/480/people/103`,
      `https://placeimg.com/640/480/people/104`,
    ],
    rating: 3,
    text: `We would like to thank all the staff, who were very friendly during our stay.`,
    date: `10.10.2019`,
  },

  {
    id: 2,
    name: `Mary`,
    avatar: [`https://placeimg.com/640/480/people/101`,
      `https://placeimg.com/640/480/people/102`,
      `https://placeimg.com/640/480/people/103`,
      `https://placeimg.com/640/480/people/104`,
    ],
    rating: 2,
    text: `The hotel had everything we needed and it was in an excellent location, being in a lovely area with lots of places to enjoy ourselves.`,
    date: `10.09.2019`,
  },

  {
    id: 3,
    name: `Oliver`,
    avatar: [`https://placeimg.com/640/480/people/101`,
      `https://placeimg.com/640/480/people/102`,
      `https://placeimg.com/640/480/people/103`,
      `https://placeimg.com/640/480/people/104`,
    ],
    rating: 5,
    text: `Iâ€™ve been many times and I will come again and again.`,
    date: `10.09.2020`,
  },


];
