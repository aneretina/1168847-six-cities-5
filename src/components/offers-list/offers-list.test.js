import React from "react";
import renderer from 'react-test-renderer';
import {BrowserRouter} from 'react-router-dom';
import {createStore} from 'redux';
import {Provider} from 'react-redux';
import OffersList from "./offers-list";
import rootReducer from "../../store/reducers/root/root-reducer";
import {AuthorizationStatus} from "../../const";
import {offers} from "../../mocks/offers";


const store = createStore(rootReducer);
const noop = () => {};

it(`OfferList render correctly`, () => {
  const tree = renderer
    .create(
        <Provider store={store}>
          <BrowserRouter>
            <OffersList
              offers={offers}
              className={`cities__places-list tabs__content`}
              authorizationStatus={AuthorizationStatus.NOT_AUTHORIZED}
              changeFavoriteStatusAction={noop}
            />
          </BrowserRouter>
        </Provider>
    ).toJSON();

  expect(tree).toMatchSnapshot();
});
