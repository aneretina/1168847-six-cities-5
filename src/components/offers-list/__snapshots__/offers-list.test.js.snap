// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OfferList render correctly 1`] = `
<div
  className="places__list cities__places-list tabs__content"
>
  <article
    className="cities__place-card place-card"
    id={15}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <div
      className="place-card__mark"
    >
      <div
        className="place-card__mark"
      >
        <span>
          Premium
        </span>
      </div>
    </div>
    <div
      className="cities__image-wrapper place-card__image-wrapper"
    >
      <a
        href="/offer/15"
        onClick={[Function]}
      >
        <img
          alt="Place image"
          className="place-card__image"
          height="200"
          src="img/apartment-01.jpg"
          width="260"
        />
      </a>
    </div>
    <div
      className="place-card__info"
    >
      <div
        className="place-card__price-wrapper"
      >
        <div
          className="place-card__price"
        >
          <b
            className="place-card__price-value"
          >
            €
            180
          </b>
          <span
            className="place-card__price-text"
          >
            / night
          </span>
        </div>
        <button
          className="place-card__bookmark-button place-card__bookmark-button--active button"
          onClick={[Function]}
          type="button"
        >
          <svg
            className="place-card__bookmark-icon"
            height={19}
            width={18}
          >
            <use
              xlinkHref="#icon-bookmark"
            />
          </svg>
          <span
            className="visually-hidden"
          >
            In bookmarks
          </span>
        </button>
      </div>
      <div
        className="place-card__rating rating"
      >
        <div
          className="place-card__stars rating__stars"
        >
          <span
            style={
              Object {
                "width": "60%",
              }
            }
          />
          <span
            className="visually-hidden"
          >
            Rating
          </span>
        </div>
      </div>
      <h2
        className="place-card__name"
      >
        <a
          href="/offer/15"
          onClick={[Function]}
        >
          Wood and stone place
        </a>
      </h2>
      <p
        className="place-card__type"
      >
        room
      </p>
    </div>
  </article>
  <article
    className="cities__place-card place-card"
    id={2}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <div
      className="place-card__mark"
    >
      <div
        className="place-card__mark"
      >
        <span>
          Premium
        </span>
      </div>
    </div>
    <div
      className="cities__image-wrapper place-card__image-wrapper"
    >
      <a
        href="/offer/2"
        onClick={[Function]}
      >
        <img
          alt="Place image"
          className="place-card__image"
          height="200"
          src="img/apartment-02.jpg"
          width="260"
        />
      </a>
    </div>
    <div
      className="place-card__info"
    >
      <div
        className="place-card__price-wrapper"
      >
        <div
          className="place-card__price"
        >
          <b
            className="place-card__price-value"
          >
            €
            80
          </b>
          <span
            className="place-card__price-text"
          >
            / night
          </span>
        </div>
        <button
          className="place-card__bookmark-button place-card__bookmark-button--active button"
          onClick={[Function]}
          type="button"
        >
          <svg
            className="place-card__bookmark-icon"
            height={19}
            width={18}
          >
            <use
              xlinkHref="#icon-bookmark"
            />
          </svg>
          <span
            className="visually-hidden"
          >
            In bookmarks
          </span>
        </button>
      </div>
      <div
        className="place-card__rating rating"
      >
        <div
          className="place-card__stars rating__stars"
        >
          <span
            style={
              Object {
                "width": "80%",
              }
            }
          />
          <span
            className="visually-hidden"
          >
            Rating
          </span>
        </div>
      </div>
      <h2
        className="place-card__name"
      >
        <a
          href="/offer/2"
          onClick={[Function]}
        >
          Beautiful & luxurious apartment in Paris
        </a>
      </h2>
      <p
        className="place-card__type"
      >
        Apartment
      </p>
    </div>
  </article>
  <article
    className="cities__place-card place-card"
    id={9}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <div
      className="place-card__mark"
    >
      
    </div>
    <div
      className="cities__image-wrapper place-card__image-wrapper"
    >
      <a
        href="/offer/9"
        onClick={[Function]}
      >
        <img
          alt="Place image"
          className="place-card__image"
          height="200"
          src="img/apartment-03.jpg"
          width="260"
        />
      </a>
    </div>
    <div
      className="place-card__info"
    >
      <div
        className="place-card__price-wrapper"
      >
        <div
          className="place-card__price"
        >
          <b
            className="place-card__price-value"
          >
            €
            220
          </b>
          <span
            className="place-card__price-text"
          >
            / night
          </span>
        </div>
        <button
          className="place-card__bookmark-button button"
          onClick={[Function]}
          type="button"
        >
          <svg
            className="place-card__bookmark-icon"
            height={19}
            width={18}
          >
            <use
              xlinkHref="#icon-bookmark"
            />
          </svg>
          <span
            className="visually-hidden"
          >
            To bookmarks
          </span>
        </button>
      </div>
      <div
        className="place-card__rating rating"
      >
        <div
          className="place-card__stars rating__stars"
        >
          <span
            style={
              Object {
                "width": "20%",
              }
            }
          />
          <span
            className="visually-hidden"
          >
            Rating
          </span>
        </div>
      </div>
      <h2
        className="place-card__name"
      >
        <a
          href="/offer/9"
          onClick={[Function]}
        >
          Beautiful & luxurious apartment in Cologne
        </a>
      </h2>
      <p
        className="place-card__type"
      >
        Apartment
      </p>
    </div>
  </article>
</div>
`;
